pkg/virt-config/config-map.go

```diff
@@ -202,6 +212,16 @@ func setConfig(config *Config, configMap *k8sv1.ConfigMap) error {
 		}
 	}
 
+	// set smbios values if they exist
+	smbiosConfig := strings.TrimSpace(configMap.Data[SmbiosConfigKey])
+	if smbiosConfig != "" {
+		// only set values if they were specified, default  values stay intact
+		err := yaml.NewYAMLOrJSONDecoder(strings.NewReader(smbiosConfig), 1024).Decode(config.SmbiosConfig)
```

Comment:
This will be an issue as `config.SmbiosConfig` is currently `nil`. It has been initialized in the default config on line 127.

From my pov, it's better to have a default set. However, if you prefer not to have a default, you then you'll need to check `config.SmbiosConfig` is not nil and if it is you need to parse the `smbiosConfig` and set it, something like:

                        var newSMBconfig *SmbiosConfig
			err := json.Unmarshal([]byte(smbiosConfig), &newSMBconfig)
1024).Decode(newSMBconfig)
			if err != nil {
				return fmt.Errorf("somehting")
			}
			config.SmbiosConfig = newSMBconfig

Reply:
Actually, @fabiand had suggested on the card that config should Default to ("KubeVirt", "KubeVirt", None), so adding it as part of `defaultClusterConfig`